src/business_logic/fact_state/in_memory_state_reader.rs

```diff
@@ -74,13 +86,37 @@ impl StateReader for InMemoryStateReader {
     fn count_actual_storage_changes(&mut self) -> (usize, usize) {
         todo!()
     }
+
+    fn get_compiled_class(
+        &mut self,
```

Comment:
Why is this `mut`? The methods below don't look, at least name wise, like they need to mutate anything.